jamaljsr/polar
# feat(docker): support out of band node releases

### Description

This PR adds the ability for Polar to use updated docker image versions in-between app releases. There are currently new versions of LND, c-lightning & bitcoind available, but they are not available inside Polar. After the next release, we'll be able to make new images available and allow users to use these images without even needing an app restart.

There is a new metadata file stored in /docker/nodes.json which contains information about which images are stored on Docker Hub. In addition, there is some compatibility information which also needs to be specified for new LN node releases.

### Privacy Note 
I thought about making the *Check for Updates* automatic but I know some users don't like apps that contact third-party servers without permission. So I made it explicit in this first implementation. I also fetch directly from the Github repo so anyone can see exactly the what data is being pulled. A future improvement could be to allow the user to decide if they want automatic checking for updates or manual. 

### Steps to Test

1. Create a Network or view an existing Network
1. In the Network Designer sidebar, click on the **Show All Versions** toggle
1. At the bottom of the node list, click on the **Check for new Node Versions** link
1. A dialog will open displaying the new versions available
1. Click the **Add New Versions** button to begin using them

### Screenshots

Before:
![image](https://user-images.githubusercontent.com/1356600/71599151-a32b7d80-2b17-11ea-9e37-c49eaf4099b6.png)

Check for Updates...
![image](https://user-images.githubusercontent.com/1356600/71599180-bdfdf200-2b17-11ea-9990-aa00962050c4.png)

After:
![image](https://user-images.githubusercontent.com/1356600/71599214-de2db100-2b17-11ea-9843-00dbff9afd63.png)

![image](https://user-images.githubusercontent.com/1356600/71599217-e4bc2880-2b17-11ea-8ea7-1887363507b3.png)
